src/test/kotlin/DirectoryLogicTester.kt

```diff
@@ -0,0 +1,52 @@
+import com.github.doyaaaaaken.kotlincsv.dsl.csvReader
+import org.junit.Test
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.assertThrows
+import java.io.File
+import java.lang.IllegalArgumentException
+
+class DirectoryLogicTester {
+
+    private val directoryLogic = DirectoryLogic()
+
+    @Test
+    fun testBuildFullFilePath() {
+
+        var testFile: File = File("D:\\Projects\\SADirectoryMaker\\src\\test\\resources\\TestCases.csv")
```

Comment:
You're committing this file but the test will fail if anyone just clones this repo and runs the tests because you have hardcoded that this test must be run on windows, that the repo must be cloned to /Projects/SADirectoryMaker, and that the repo must be cloned to the D:\ drive.

You want a relative path here.

Or, better yet, because you placed the file in resources, you can read the file out of your resources.